packages/p2p/app/index.js

```diff
@@ -0,0 +1,68 @@
+/* P2P peer discovery server */
+
+const WSServer = require('rpc-websockets').Server
+
+function conf (name, defaultValue) {
+  var value = process.env[name]
+  if (value === undefined) {
+    value = defaultValue
+  }
+  if (value === undefined) {
+    throw new Error(`Required configuration "${name}" not found.`)
+  }
+  return value
+}
+
+const host = conf('PEER_DISCOVERY_HOST', 'localhost')
+const port = conf('PEER_DISCOVERY_PORT', '8889')
+
+const server = new WSServer({ host, port })
+
+console.log(`Peer discovery server listening on ws://${host}:${port}`)
+
+const ROOMS = {}
+
+server.register('joinRoom', ([room, id]) => {
+  if (!ROOMS[room]) { ROOMS[room] = [] }
+  if (ROOMS[room][id]) {
+    console.log(`Error: ID '${id}' already in peer discovery channel '${room}'`)
+    return
+  }
+  console.log(`Id ${id} joining room ${room}`)
+  ROOMS[room].push(id)
+
+  // Time out from the room after 1 minute
+  setTimeout(() => {
+    console.log(`ID ${id} timed out from room ${room}`)
+    leaveRoom(room, id)
+  }, 240000)
+})
+
+server.register('leaveRoom', ([room, id]) => {
+  console.log(`ID ${id} leaving room ${room}`)
+  leaveRoom(room, id)
+})
+
+server.register('rollCall', ([room]) => {
+  if (ROOMS[room]) {
+    return ROOMS[room]
+  }
+  return []
+})
+
+server.register('addEvent', ([event]) => {
```

Comment:
Why is this event coming in as a list?

Reply:
This is was the way it was presented in the documentation for rpc-websockets -- I'll check if it works without the list.

Reply:
Yup: `TypeError: "params" must be an object or "Array"`